workflow/realisations.py

```diff
@@ -0,0 +1,676 @@
+            ): And(float, is_positive),
+            Literal(
+                "genslip_seed", description="The random seed passed to genslip"
+            ): And(int, is_non_negative),
+            Literal("genslip_version", description="The version of genslip to use"): Or(
+                "5.4.2"
+            ),
+            Literal(
+                "srfgen_seed",
+                description="A second random seed for genslip (TODO: how does genslip use this value?)",
+            ): And(int, is_non_negative),
+        }
+    ),
+    DomainParameters: Schema(
+        {
+            Literal("resolution", description="The simulation resolution (in km)"): And(
+                float, is_positive
+            ),
+            Literal(
+                "centroid", description="The centroid location of the model"
+            ): LAT_LON_SCHEMA,
+            Literal("width", description="The width of the model (in km)"): And(
+                float, is_positive
+            ),
+            Literal("length", description="The length of the model (in km)"): And(
+                float, is_positive
+            ),
+            Literal("depth", description="The depth of the model (in km)"): And(
+                float, is_positive
+            ),
+        }
+    ),
+    RupturePropagationConfig: Schema(
+        {
+            str: {
+                Literal(
+                    "parent",
+                    description="The parent fault that triggers this fault (or null if the initial fault)",
+                ): Or(str, None),
+                Literal(
+                    "hypocentre",
+                    description="The hypocentre coordinates (or initial rupture point if not the initial fault)",
+                ): FAULT_LOCAL_COORDINATES_SCHEMA,
+                Literal(
+                    "magnitude",
+                    description="The total moment magnitude for the rupture on this fault",
+                ): And(float, is_plausible_magnitude),
+                Literal("rake", description="The fault rake"): And(
+                    float, is_valid_degrees
+                ),
+            }
+        }
+    ),
+    RealisationMetadata: Schema(
+        {
+            Literal("name", description="The name of the realisation"): str,
+            Literal("version", description="The version of the realisation format"): Or(
+                "1"
+            ),
+        }
+    ),
+}
+
+_REALISATION_KEYS = {
+    SourceConfig: "sources",
+    SRFConfig: "srf",
+    DomainParameters: "domain",
+    RupturePropagationConfig: "rupture_propagation",
+    RealisationMetadata: "metadata",
+}
+
+
+LoadableConfig = Union[
+    SourceConfig,
+    SRFConfig,
+    DomainParameters,
+    RupturePropagationConfig,
+    RealisationMetadata,
+]
+
+
+class RealisationParseError(Exception):
+    pass
+
+
+def read_config_from_realisation(config: type, realisation_ffp: Path) -> LoadableConfig:
```

Comment:
Is there any way to load all of the different parameters at once without needing to call this multiple times for each of the different subsections?

Reply:
Good point @joelridden, makes me curious too!

Reply:
So the short answer is no. I have thought about this a bit. The truth is that I don't think there is a massive need for it. No part of the workflow needs to know about the SRF generation parameters and the HF parameters, for example. However, there is some scope to extend this by, say, making `config` optionally a list and then returning a list of `LoadableConfig` in that case. I'll write the code for that now.